docs/getting-started.md

```diff
@@ -0,0 +1,117 @@
+
+The protos are located in a github repo: https://github.com/namely/chief-of-state-protos
+
+A popular way of incorporating them into your project is via git submodules,
+which you can add like so:
+```sh
+# adds COS protos to proto/cos
+git submodule add git@github.com:namely/chief-of-state-protos ./protos/cos
+```
+
+From there, you can generate the COS interfaces for your language (see the
+official quick start guide for your language https://grpc.io/docs/languages/).
+
+As an example, one would use [protoc](https://grpc.io/docs/protoc-installation/)
+to generate java source code like so:
+```sh
+export SRC_DIR=protos/cos
+export DST_DIR=build/gen
+# generate everything in protos/cos/chief_of_state/v1/common.proto
+protoc -I=$SRC_DIR --java_out=$DST_DIR $SRC_DIR/foo.proto
+# generate ALL .proto files in protos/cos
+find $SRC_DIR -name "*.proto" | xargs -I{} protoc -I=$SRC_DIR --java_out=$DST_DIR {}
+```
+
+**Pro tip:** many languages have wrappers around protoc to make this easier,
+like [scalap-pb](https://scalapb.github.io/) for scala (which chief-of-state
+uses internally). Namely also offers an open-source docker-based solution
+[namely/docker-protoc](https://github.com/namely/docker-protoc) with support
+for many languages.
+
+## Implement a "Write Side"
+
+Drawing from CQRS terminology, COS defines a [writeside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/writeside.proto) that you must implement to define how state
+is created and updated.
+
+### Methods
+`HandleCommand` accepts a command (think "request") and a prior state and returns an event.
+For example, given a command to `UpdateUserEmail` and a User, this RPC might return
+an event called `UserEmailUpdated`. This method is allowed to throw exceptions
+(as gRPC error statuses) if the inbound command (request) is invalid.
+
+`HandleEvent` accepts an event and the prior state of an entity and returns a new state.
+For example, given a UserEmailUpdated event and a User, this RPC would return a
+new User instance with the email updated. It's advisable to do all your
+validations in `HandleCommand` such that this method is unlikely to encounter
+invalid data. If `HandleCommand` fails, `HandleEvent` is not called for that
+command.
+
+### Registration
+Register your write handler service as COS environment variables like so:
+
+```yaml
+# set the host and port for your write handler gRPC server
+COS_WRITE_SIDE_HOST: localhost
+COS_WRITE_SIDE_PORT: 50051
+```
+
+(If you are new to setting up COS, see the full [configuration](docs/configuration.md) guide.)
+
+### Sending requests to COS
+Once you have implemneted a write handler and registered it with COS, you are
+able to start sending requests to chief-of-state with the chief-of-state
+[gRPC client methods](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/service.proto).
+
+`ProcessCommand` accepts a command (request) and an entity ID, and processes
```

Comment:
I think we should reiterate what the entity ID is.
I like the explanation for GetState to think of this as keys and values and I think it's also useful for the context of creating and updating an entity.